### DISCUSSION TITLE
Easy way to run MongoDB?

### DISCUSSION THREAD

**Discussion by mindplay-dk** [danny-avila/LibreChat#852] [Q&A] [answered]
Is there an easy way to get a MongoDB server up and going? for someone who does not want to learn MongoDB and just wants to run the app locally, and does not want to put their personal data in somebody's cloud.

I'd also rather not use Docker, since the last time I ran it, my system became extremely unstable and I ended up having to reinstall Windows. (yeah.)

I tried the Windows installer, and figured out how to create a database, but couldn't get a connection.

I'm using WSL 2 and also tried using `mongoz` to get an instance going - it looks like there's a mongo server running under Ubuntu now, but I still can't get a connection on that port.

The error I get is `MongooseServerSelectionError: connect ECONNREFUSED 127.0.0.1:27017` after about 20 seconds, I'm assuming that's a timeout and it can't connect at all?

```
MongooseServerSelectionError: connect ECONNREFUSED 127.0.0.1:27017
    at _handleConnectionErrors (/home/mindplay/workspace/LibreChat/node_modules/mongoose/lib/connection.js:788:11)
    at NativeConnection.openUri (/home/mindplay/workspace/LibreChat/node_modules/mongoose/lib/connection.js:763:11)
    at async connectDb (/home/mindplay/workspace/LibreChat/api/lib/db/connectDb.js:40:17)
    at async startServer (/home/mindplay/workspace/LibreChat/api/server/index.js:17:3) {
  reason: TopologyDescription {
    type: 'Unknown',
    servers: Map(1) { 'localhost:27017' => [ServerDescription] },
    stale: false,
    compatible: true,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    setName: null,
    maxElectionId: null,
    maxSetVersion: null,
    commonWireVersion: 0,
    logicalSessionTimeoutMinutes: null
  },
  code: undefined
}
```

Is there a MongoDB substitute that's easier to install that just saves things in a SQLite db or something? I really don't need cloud scale or anything fancy, I just want to run the app locally.

Damn you, mongo! 😖

**Comment by danny-avila** [comment]
Yes in short, mongodb atlas. Free and remote and works well with the project. I can see if anyone did a write up for setting it up

**Comment by danny-avila** [comment, marked as answer]
Yup, there is a write-up that will be useful for you! https://docs.librechat.ai/install/mongodb.html#create-an-account

**Comment by mindplay-dk** [comment]
I hopped in and fixed `mongoz` instead 🙂

https://github.com/unjs/mongoz/issues/4

for someone just drive-by testing this, or for non-developers and end-users who just want a local install, there's really no reason why any of this should require more than a simple command like `npx mongoz` and then off you go.

the package has an API as well, by the way - I mean, wouldn't it be nice if you could literally just clone the repo and launch the whole thing with `npm start` without any further ado? 🙂

I could even see turning this into a one-click install with electron or something? for the completely clueless 😅

**Reply by danny-avila** [reply]
A one-click install would be great! I was not aware of mongoz, that looks really handy

**Reply by mindplay-dk** [reply]
yeah, it was hard coded to install a specific version of Mongo, which didn't work anymore since Ubuntu 22.04 - that might have limited it's success. I don't know yet if the maintainer still cares, we'll see if he responds - he hasn't touched the project in 2 years, so it might be dead. If needed, I might fork the project and publish it under a different name. Let's give him a week or so. :-)
